src/System.Net.Primitives/src/System/Net/Cookie.cs

```diff
@@ -0,0 +1,1755 @@
+                                        try
+                                        {
+                                            cookie.Port = _tokenizer.Value;
+                                        }
+                                        catch (CookieException)
+                                        {
+                                            // this cookie will be rejected
+                                            cookie.InternalSetName(string.Empty);
+                                        }
+                                    }
+                                    break;
+
+                                case CookieToken.Version:
+                                    // this is a new cookie, this token is for the next cookie.
+                                    _savedCookie = new Cookie();
+                                    int parsed;
+                                    if (int.TryParse(_tokenizer.Value, out parsed))
+                                    {
+                                        _savedCookie.Version = parsed;
+                                    }
+                                    return cookie;
+
+                                case CookieToken.Unknown:
+                                    // this is a new cookie, the token is for the next cookie.
+                                    _savedCookie = new Cookie();
+                                    if (_savedCookie.InternalSetName(_tokenizer.Name) == false)
+                                    {
+                                        // will be rejected
+                                        _savedCookie.InternalSetName(string.Empty);
+                                    }
+                                    _savedCookie.Value = _tokenizer.Value;
+                                    return cookie;
+                            }
+                            break;
+
+                        case CookieToken.Attribute:
+                            switch (_tokenizer.Token)
+                            {
+                                case CookieToken.Port:
+                                    if (!portSet)
+                                    {
+                                        portSet = true;
+                                        cookie.Port = string.Empty;
+                                    }
+                                    break;
+                            }
+                            break;
+                    }
+                }
+            } while (!_tokenizer.Eof && !_tokenizer.EndOfCookie);
+            return cookie;
+        }
+
+        internal static string CheckQuoted(string value)
+        {
+            if (value.Length < 2 || value[0] != '\"' || value[value.Length - 1] != '\"')
+                return value;
+
+            return value.Length == 2 ? string.Empty : value.Substring(1, value.Length - 2);
+        }
+    }
+
+    internal class Comparer : IComparer
```

Comment:
And since this isn't nested inside of ```Cookie```, should it be named ```CookieComparer``` or something more specific than just ```Comparer```?